readthedocs/core/views/hooks.py

```diff
@@ -105,30 +105,29 @@ def get_or_create_external_version(project, identifier, verbose_name):
     :returns:  External version.
     :rtype: Version
     """
-    external_version = project.versions(
-        manager=EXTERNAL
-    ).filter(verbose_name=verbose_name).first()
+    external_version, created = project.versions.get_or_create(
+        verbose_name=verbose_name,
+        type=EXTERNAL,
+        defaults={'identifier': identifier, 'active': True},
+    )
 
-    if external_version:
+    if created:
+        log.info(
+            '(Create External Version) Added Version: [%s] ', ' '.join(
+                external_version.slug
```

Comment:
Why are we calling `join` on a slug?

Reply:
Opps! I was doing something else forgot to remove it sorry.